File: 0007-reverse-integer/0007-reverse-integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        sign = -1 if x<0 else 1
        x = abs(x)
        rever = 0
        while(x!=0) :
            digit = x % 10
            rever = (rever*10) + digit
            x //= 10
        rever *= sign
        if rever < -2**31 or rever > 2**31 - 1:
            return 0
            
        return rever

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int digit;
        int reversed = 0;

        while (x != 0) {
            digit = x % 10;

            if (reversed > Integer.MAX_VALUE / 10 || reversed < Integer.MIN_VALUE / 10) {
                return 0;
            }

            reversed = (reversed * 10) + digit;
            x = x / 10;
        }
        return reversed;
    }
}